src/pieces/King.js

```diff
@@ -22,27 +22,34 @@ class King extends Piece {
   }
 
   findLegalMoves(gameArea) {
+    const possibleMoves = [];
+    const attack = this.findAttackingMoves(gameArea).flat();
```

Comment:
To `.flat()` nic nie wnosi. Tablica zwracana przez findAttackingMoves nie jest zagnieżdżona